src-relution/mwListCollectionFilter.js

```diff
@@ -22,6 +26,66 @@ angular.module('mwCollection')
           property: null
         };
 
+      var preventRouteReload = function () {
+        //Check whether reloadOnSearch is already disabled
+        if ($route.current.$$route.reloadOnSearch === false) {
+          return;
+        }
+        // Remember the state so we can set it to the original state after we have updated the route
+        var prevReloadOnSearchVal = $route.current.$$route.reloadOnSearch;
+        //Set reloadOnSearch false so angular does not reinitialize the controller
+        $route.current.$$route.reloadOnSearch = false;
+        //Route update is triggered when reloadOnSearch is set to true and a search param has changed
+        var off = $rootScope.$on('$routeUpdate', function () {
+          $route.current.$$route.reloadOnSearch = prevReloadOnSearchVal;
+          off();
```

Comment:
Why do we need self calling in here?
Better naming for this method would be "better"